scripts/metadata_comparison/extractor.py

```diff
@@ -0,0 +1,193 @@
+#   * pip3 install --upgrade google-api-python-client
+#
+# Remember to login to create application default credentials before use:
+#   % gcloud auth application-default login
+
+import argparse
+import re
+import json
+import requests
+from google.cloud import storage
+import google.auth
+from apiclient.discovery import build
+
+def workflow_regex_validator(value):
+    workflow_regex=re.compile('^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$')
+    if not workflow_regex.match(value):
+        msg = 'Invalid workflow ID {inv_id}. Expected {regex}'.format(inv_id=value, regex=workflow_regex.pattern)
+        raise argparse.ArgumentTypeError(msg)
+    else:
+        return value
+
+
+def url_regex_validator(value):
+    url_regex = re.compile('(http(s?)://.*[^/])(/(api.*)?)?')
+    m = url_regex.search(value)
+    if m:
+        return m.group(1)
+    else:
+        msg = 'Invalid Cromwell URL {inv_url}. Expected {regex}'.format(inv_url=value, regex=url_regex.pattern)
+        raise argparse.ArgumentTypeError(msg)
+
+
+def gcs_path_regex_validator(value):
+    gcs_regex = re.compile('^gs://([a-zA-Z0-9-]+)/(([a-zA-Z0-9-]+/)*[a-zA-Z0-9-]+)/?$')
+    m = gcs_regex.search(value)
+    if m:
+        return m.group(1), m.group(2)
+    else:
+        msg = 'Invalid GCS path {inv_gcs}. Expected {regex}'.format(inv_gcs=value, regex=gcs_regex.pattern)
+        raise argparse.ArgumentTypeError(msg)
+
+
+def get_operation_id_number(value):
+    """Extracts from operation IDs like 'projects/project_name/operations/01234567891011121314' just the final
+    number - eg '01234567891011121314'"""
+
+    operation_regex = re.compile('^.*/([^/]*)')
+    m = operation_regex.search(value)
+    if m:
+        return m.group(1)
+    else:
+        msg = 'Unexpected operation ID {inv_url}. Expected something like {regex}'.format(inv_url=value, regex=url_regex.pattern)
+        raise Exception(msg)
+
+
+def uploadLocalCheckout():
+    #   # Make a snapshot of the local Cromwell git repo
+    #
+    #   hash=$(git rev-parse HEAD)
+    #
+    #   # Make sure all files are checked in.
+    #   if [[ $(git status --short | wc -l) -ne 0 ]]; then echo "stuff needs to be checked in"; exit 1; fi
+    #
+    #   git_snapshots_dir="../git-snapshots"
+    #   mkdir -p "${git_snapshots_dir}"
+    #   git_snapshot="${git_snapshots_dir}/$hash"
+    #   git clone . --local "${git_snapshot}"
+    raise Exception("Not Implemented")
+
+
+def fetch_raw_workflow_metadata(cromwell_url, workflow):
+    url = '{url}/api/workflows/v1/{wfid}/metadata?expandSubWorkflows=true'.format(url=cromwell_url, wfid=workflow)
+    print('Fetching Cromwell metadata from...', url, end='...')
+    result = requests.get(url)
+    print(' fetched!')
+    return result.content, result.json()
+
+
+def find_operation_ids_in_metadata(json_metadata):
+    """Finds all instances of PAPI operations IDs in a workflow, and the API to call to retrieve metadata"""
```

Comment:
Seems like a case where we're better off storing everything and letting the digester decide which ones it cares about, rather than deciding that at extractor time?